%\VignetteIndexEntry{About diffcoexp}
\documentclass{article}
\usepackage{authblk}
\title{About diffcoexp}
\date{2018-01-30}
\author{Wenbin Wei, Sandeep Amberkar, Winston Hide}
\affil{Sheffield Institute of Translational Neuroscience, University of Sheffield, Sheffield, United Kingdom}
\usepackage[utf8]{inputenc}
\begin{document}
\SweaveOpts{concordance=TRUE}
\maketitle
\tableofcontents
\section{Description}
This package identifies differentially coexpressed links (DCLs) and differentially 
coexpressed genes (DCGs). DCLs are gene pairs with significantly different correlation 
coefficients under two conditions (de la Fuente 2010, Jiang et al., 2016). DCGs are genes 
with significantly more DCLs than by chance (Yu et al., 2011, Jiang et al., 2016). It 
takes two gene expression matrices or data frames under two conditions as input, 
calculates gene-gene correlations under two conditions and compares them with Fisher's Z 
transformation. It filters gene pairs with the thresholds for correlation coefficients 
and their adjusted p value as well as the thresholds for the difference between the two 
correlation coefficients and its adjusted p value. It identifies DCGs using binomial 
probability model (Jiang et al., 2016).\smallskip

\noindent
The main steps are as follows:

a). Correlation coefficients and p values of all gene pairs under two conditions are 
calculated.

b). The difference between the correlation coefficients under two conditions are 
calculated and the p value is calculated using Fisher's Z-transformation.

c). p values are adjusted.

d). Gene pairs (links) coexpressed in at least one condition are identified using the 
criteria that at least one of the correlation coefficients under two conditions having 
absolute value greater than the threshold rth and the adjusted p value less than the 
threshold qth. The links that meet the criteria are included in CLs.

e). Differentially coexpressed gene pairs (links) are identified from CLs using the 
criteria that the absolute value of the difference between the two correlation 
coefficients is greater than the threshold r.diffth and the adjusted p value is less than 
the threshold q.diffth. The links that meet the criteria are included in DCLs.

f). The DCLs are classified into three categories: "same signed", "diff signed", or 
"switched opposites". "same signed" indicates that the gene pair has same signed 
correlation coefficients under both conditions. "diff signed" indicates that the gene 
pair has oppositely signed correlation coefficients under two conditions and only one of 
them meets the criteria that the absolute correlation coefficient is greater than the 
threshold rth and adjusted p value less than the threshold qth. "switched opposites" 
indicates that the gene pair has oppositely signed correlation coefficients under two 
conditions and both of them meet the criteria that the absolute correlation coefficients 
are greater than the threshold rth and adjusted p value less than the threshold qth.

g). All the genes in DCLs are tested for their enrichment of DCLs, i.e, whether they have 
more DCLs than by chance using binomial probability model (Jiang et al., 2016). Those 
with adjusted p value less than the threshold q.dcgth are included in DCGs.
\section{Installation and removal}
\noindent
To install this package, start R and enter: \par
library(devtools) \par
devtools::install\_git("git://github.com/hidelab/diffcoexp.git", \par
    branch = "master") \smallskip
\par
\noindent
The above method does not build and install vignette. To install the package with 
vignette, enter the following from command line: \par
\par
git clone https://github.com/hidelab/diffcoexp.git \par
R CMD build diffcoexp \par
R CMD check diffcoexp\_0.99.0.tar.gz \par
R CMD INSTALL diffcoexp\_0.99.0.tar.gz \smallskip
\par
\noindent
To remove this package, start R and enter: \par
\par
@
remove.packages("diffcoexp")
\section{Example}
This example illustrates the identification of differentially coexpressed links (gene 
pairs) and differentially coexpressed genes of yeast after pulses of 2 g/l and 0.2 g/l 
glucose separately. The data were downloaded from GEO (GSE4158). Only 200 genes were 
analysed in this example. Analysis of all the genes (6104) will take about 20 minutes on 
a computer with 8 cores and 16GB RAM.
<<results=hide>>=
library(diffcoexp)
data(gse4158part)
@
<<>>=
allowWGCNAThreads()
res=diffcoexp(exprs.1 = exprs.1, exprs.2 = exprs.2, r.method = "spearman" )
@
The results are a list of two data frames, one for differentially co-expressed links 
(DCLs, gene pairs), one for differentially co-expressed genes (DCGs).
<<>>=
str(res)
@
\section{References}
1. de la Fuente A. From “differential expression” to “differential networking” – 
identification of dysfunctional regulatory networks in diseases. Trends in Genetics. 
2010 Jul;26(7):326-33.

\noindent
2. Jiang Z, Dong X, Li Z-G, He F, Zhang Z. Differential Coexpression Analysis Reveals 
Extensive Rewiring of Arabidopsis Gene Coexpression in Response to Pseudomonas syringae 
Infection. Scientific Reports. 2016 Dec;6(1):35064.

\noindent
3. Yu H, Liu B-H, Ye Z-Q, Li C, Li Y-X, Li Y-Y. Link-based quantitative methods to 
identify differentially coexpressed genes and gene pairs. BMC bioinformatics. 
2011;12(1):315.
\end{document}
